crates/grit-pattern-matcher/src/pattern/state.rs

```diff
@@ -37,7 +37,7 @@ pub struct FileRegistry<'a, Q: QueryContext> {
     /// The number of versions for each file
     version_count: Vec<u16>,
     /// Original file paths, for lazy loading
-    file_paths: Vec<&'a PathBuf>,
+    file_paths: Vec<&'a Path>,
```

Comment:
Changing it to a `&Path` makes fewer assumptions on where the paths come from, and it doesn't appear to have any downstream effects.